src/components/chartArea/chartArea.js

```diff
@@ -169,67 +111,169 @@ class ChartArea extends React.Component {
+        dataSet.thresholds.forEach(value => {
+          dataSetMaxY = value.y > dataSetMaxY ? value.y : dataSetMaxY;
+        });
+      }
+    });
+
+    if (!isXAxisTicks) {
+      generatedDomain.x = [0, dataSetMaxX || 10];
     }
 
-    if (Object.keys(domain).length) {
-      chartProps.domain = domain;
+    if (!isYAxisTicks) {
+      const floored = Math.pow(10, Math.floor(Math.log10(dataSetMaxY || 10)));
+      generatedDomain.y = [0, Math.ceil((dataSetMaxY + 1) / floored) * floored];
+    }
+
+    if (Object.keys(generatedDomain).length) {
+      updatedChartDomain.domain = generatedDomain;
+    }
+
+    return {
+      maxY: dataSetMaxY,
+      chartDomain: { ...updatedChartDomain }
+    };
+  }
+
+  getChartLegend() {
+    const { dataSets } = this.props;
+    const legendData = [];
+
+    dataSets.forEach(dataSet => {
+      if (dataSet.legendThreshold) {
+        legendData.push({ symbol: { type: 'dash' }, ...dataSet.legendThreshold });
+      }
+
+      if (dataSet.legendData) {
+        legendData.push(dataSet.legendData);
+      }
+    });
+
+    return {
+      legendData,
+      legendOrientation: 'horizontal',
+      legendPosition: 'bottom'
+    };
+  }
+
+  render() {
+    const { chartWidth } = this.state;
+    const { dataSets, padding } = this.props;
+
+    const { isXAxisTicks, isYAxisTicks, xAxisProps, yAxisProps } = this.getChartTicks();
+    const { chartDomain, maxY } = this.getChartDomain({ isXAxisTicks, isYAxisTicks });
+    const chartLegendProps = this.getChartLegend();
+    const chartProps = { padding, ...chartLegendProps, ...chartDomain };
+
+    // FixMe: Check maxY has value, and conditionally apply ChartVoronoiContainer to avoid a massive memory leak?
+    if (maxY > 0) {
+      chartProps.containerComponent = <ChartVoronoiContainer labels={d => d.tooltip} />;
     }
 
     return (
       <div ref={this.containerRef}>
         <Chart width={chartWidth} {...chartProps}>
-          <ChartAxis {...updatedXAxisProps} />
-          <ChartAxis {...updatedYAxisProps} />
-          {(dataSetOne.thresholds && dataSetOne.thresholds.length && (
-            /** fixme: split this out into a new wrapper called ChartThreshold in PF React */
-            <ChartLine data={dataSetOne.thresholds} style={{ data: { strokeDasharray: 3.3 } }} />
-          )) ||
+          <ChartAxis {...xAxisProps} />
+          <ChartAxis {...yAxisProps} />
+          {(dataSets &&
+            dataSets.length &&
+            dataSets.map(
+              dataSet =>
+                (dataSet.thresholds && dataSet.thresholds.length && (
+                  /** fixme: split this out into a new wrapper called ChartThreshold in PF React */
+                  <ChartLine
+                    interpolation={dataSet.thresholdInterpolation || 'step'}
+                    key={helpers.generateId()}
+                    data={dataSet.thresholds}
+                    style={dataSet.thresholdStyle || { data: { strokeDasharray: 3.3 } }}
```

Comment:
work is still progressing on a Horizontal Indicator. For now, this totally works. I think the current suggestion is to use a `strokeDasharray` of `6,6` - i'm planning to update the PF PR today...
https://github.com/patternfly/patternfly-react/pull/2796

Reply:
obviously no rush on adjusting that style now